File: samples/CslaLight/cs/RemotePortalWithObjectFactory/Sample.Business.Server/Security/SampleIdentity.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.DataPortalClient;
using System.ComponentModel;

#if !SILVERLIGHT
using System.Data.SqlClient;
#endif


namespace Sample.Business
{
  [Serializable(), Csla.Server.ObjectFactory("Sample.Business.SampleIdentityFactory, Sample.Business", "", "GetSampleIdentity", "", "")]
  public class SampleIdentity : CslaIdentity
  {

#if SILVERLIGHT

    public static void GetIdentity(string username, string password, EventHandler<DataPortalResult<SampleIdentity>> completed)
    {
      GetCslaIdentity<SampleIdentity>(completed, new CredentialsCriteria(username, password));
    }

#else

	  internal void LoadData(string userName, MobileList<string> rolesList)
	  {
		AuthenticationType = "Csla";
		Roles = rolesList;
		Name = userName;
		IsAuthenticated = true;
	  }

#endif

  }

} //end of root namespace

File: samples/CslaLight/vb/RemotePortalWithObjectFactory/Sample.Business.Server/Security/SampleIdentity.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.DataPortalClient
Imports System.ComponentModel

#If Not SILVERLIGHT = 1 Then
Imports System.Data.SqlClient
#End If


<Serializable(), Csla.Server.ObjectFactory("Sample.Business.SampleIdentityFactory, Sample.Business", "", "GetSampleIdentity", "", "")> _
Public Class SampleIdentity
  Inherits CslaIdentity

#If Silverlight = 1 Then

  Public Shared Sub GetIdentity(ByVal username As String, ByVal password As String, ByVal completed As EventHandler(Of DataPortalResult(Of SampleIdentity)))
    GetCslaIdentity(Of SampleIdentity)(completed, New CredentialsCriteria(username, password))
  End Sub

#Else

  Friend Sub LoadData(ByVal userName As String, ByVal rolesList As MobileList(Of String))
    AuthenticationType = "Csla"
    Roles = rolesList
    Name = userName
    IsAuthenticated = True
  End Sub

#End If

End Class
